setup.cfg

```diff
@@ -71,3 +71,7 @@ ignore =
     sklearn/utils/_seq_dataset.pxd
     sklearn/utils/_weight_vector.pyx
     sklearn/utils/_weight_vector.pxd
+
+[codespell]
+skip = js,v*.rst,_stop_words.py
```

Comment:
There are some auto generated files that would good to skip:

```suggestion
skip = *.js,./sklearn/feature_extraction/_stop_words.py,./doc/_build,./doc/auto_examples,./doc/modules/generated
```

I think expanding `_stop_words.py` to the full path makes it more explicit. I see that `v*.rst` is skipped because of names, but I think having spell check enabled for the changelog is a net win.

Reply:
I agree about the full paths, but I'd rather wait for https://github.com/codespell-project/codespell/pull/2058 to be taken into account.

About `v*.rst`, I've left these out because many projects don't want to modify the changelog, they want typos in there to remain as they are forever. I believe this is inspired by the GNU documentation on [Change Logs](https://www.gnu.org/prep/standards/html_node/Change-Logs.html). On the other hand, not everyone agrees with the GNU documentation, after all the changelog is viewed as a detailed dump of the VCS, which may appear a bit too detailed and obsolete these days. If many of you want me to fix the typos in the changelog, I'm happy to do that. However, I suggest this happens in a distinct PR, like #21069 for the rest of the typos.

Reply:
Also, it's `js` and not `*.js`, to exclude vendored JS scripts under `doc/themes/scikit-learn-modern/static/js`, but not other JS scripts (if any in the future).